app/controllers/reviews_controller.rb

```diff
@@ -4,21 +4,36 @@ def new
   end
 
   def create
-    shelter = Shelter.find(params[:id])
-    review = shelter.reviews.create!(review_params)
-
-    redirect_to "/shelters/#{shelter.id}"
+    @shelter = Shelter.find(params[:id])
+    review = @shelter.reviews.new(review_params)
+    if review.save
+        redirect_to "/shelters/#{@shelter.id}"
+    else
+        flash.now[:notice] = "Review not created: Required information missing."
+        render :new
+    end
   end
 
   def edit
     @review = Review.find(params[:review_id])
   end
 
   def update
-    @review = Review.find(params[:review_id])
-    @review.update(review_params)
+    @shelter = Shelter.find(params[:id])
+    review = Review.find(params[:review_id])
+    if review.valid?
+      review.update!(review_params)
+      redirect_to "/shelters/#{review.shelter_id}"
+    else
+      redirect_to "/shelters/#{review.shelter_id}/#{review.id}/edit" 
+      flash[:notice] = "Review not updated: Required information missing."
```

Comment:
These flash notices will have to be formatted for usability but they're functional for now